dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java

```diff
@@ -84,38 +84,34 @@
 	private final ContentletAPI contentletAPI = APILocator.getContentletAPI();
 
 	/**
-	 * Will get a folder for you on a given path for a particular host
+	 * Will get a folder for you on a given path for a particular host.
 	 *
-	 * @param path
-	 * @param hostId
-	 * @return
+	 * If the folder does not exists will return a folder with null values.
+	 * If the user does not have permissions over the folder a DotSecurityException will be thrown.
+	 *
+	 * @param path path of the requested folder
+	 * @param hostId host id where the folder should live
+	 * @return the requested folder if the user has permissions, if not an exception.
 	 * @throws DotHibernateException
 	 */
 	private final FolderFactory folderFactory = FactoryLocator.getFolderFactory();
 	private final PermissionAPI permissionAPI = getPermissionAPI();
 
 	@CloseDBIfOpened
 	public Folder findFolderByPath(final String path, final Host host,
-								   final User user, final boolean respectFrontEndPermissions) throws DotStateException,
+			final User user, final boolean respectFrontEndPermissions) throws DotStateException,
 			DotDataException, DotSecurityException {
-
 		final Folder folder = folderFactory.findFolderByPath(path, host);
-
-		if (folder != null && InodeUtils.isSet(folder.getInode()) &&
-				!permissionAPI.doesUserHavePermission(folder, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)) {
-
-			// SYSTEM_FOLDER means if the user has permissions to the host, then they can see host.com/
-			if(FolderAPI.SYSTEM_FOLDER.equals(folder.getInode())) {
-				if(!Host.SYSTEM_HOST.equals(host.getIdentifier())){
-					if(!permissionAPI.doesUserHavePermission(host, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)) {
-						throw new DotSecurityException("User " + (user.getUserId() != null ? user.getUserId() : BLANK) + " does not have permission to read folder " + folder.getPath());
-					}
-				}
-			}
-
+		if (folder == null || UtilMethods.isEmpty(folder.getInode()) || permissionAPI.doesUserHavePermission(folder, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)){
+			return folder;
 		}
-
-		return folder;
+		if(FolderAPI.SYSTEM_FOLDER.equals(folder.getInode()) && !permissionAPI.doesUserHavePermission(host, PermissionAPI.PERMISSION_READ, user, respectFrontEndPermissions)){
+			throw new DotSecurityException("User " + (user.getUserId() != null ? user.getUserId() : BLANK) + " does not have permission to read folder " + folder.getPath() + " on host " + host.getHostname());
```

Comment:
Codacy found an issue: [The String literal "User " appears 33 times in this file; the first occurrence is on line 109](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5916479)